### ISSUE TITLE
Incremental deployment failure with 'bazel-android-intro' codelab project

### ISSUE THREAD

**Issue by oliviernotteghem** [bazelbuild/intellij#1922]
Incremental mobile deployment succeeds from command line, ie : 
`bazel mobile-install //mediarecorder/java/com/example/android/mediarecorder:mediarecorder --start_app --incremental`

However, when ran from Android Studio, several devices are specified when creating dex folder used for incremental deployment, causing command to fail : 
` ... '-s', 'emulator-5554:tcp:5037', ... `

Tried different physical / emulators device, and changing SDK version (28, 29, 30) : same issue

```
ERROR: /Users/oliviern/Uber/bazel-android-intro/solution/mediarecorder/java/com/example/android/mediarecorder/BUILD:13:15: Installing //mediarecorder/java/com/example/android/mediarecorder:mediarecorder failed (Exit 1)
Error:
adb command: ['/Users/oliviern/Library/Android/sdk/platform-tools/adb', '-s', 'emulator-5554', '-s', 'emulator-5554:tcp:5037', '-s', 'emulator-5554', '-s', 'emulator-5554:tcp:5037', 'shell', 'mkdir -p /data/local/tmp/incrementaldeployment/com.example.android.mediarecorder/dex']
return code: 1
stdout: 
stderr: adb: unknown host service
```

**Comment by mai93** [comment]
Hello @oliviernotteghem do you still see this issue with the latest version of the plugin?

**Comment by pswaminathan** [comment]
Hi @mai93, I am seeing this same issue (different project, but same behavior) with the latest plugin version.

**Comment by Morfly** [comment]
Hi @pswaminathan. Could you click `Edit Configurations...` in Idea/AndroidStudio -> select your target -> disable `use mobile-install` checkmark and try again. It did the trick for me.

**Comment by pswaminathan** [comment]
@Morfly unchecking it doesn't work either. I then get this:

```
Command: bazel build --tool_tag=ijwb:AndroidStudio --output_groups=+android_deploy_info "" --curses=no --color=yes --progress_in_terminal_title=no --fat_apk_cpu=arm64-v8a,armeabi-v7a --build_event_binary_file=/var/folders/71/0r_ylvb13vv2my1_676sx1gm0000gn/T/intellij-bep-a80e09e0-223d-4caa-ad57-b02a6ac7539d --nobuild_event_binary_file_path_conversion -- //src/seaborn/app:app

Loading: 
Loading: 0 packages loaded
ERROR: Skipping '': the empty string is not a valid target
ERROR: the empty string is not a valid target
INFO: Elapsed time: 0.345s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
FAILED: Build did NOT complete successfully (0 packages loaded)
INFO: Build Event Protocol files produced successfully.
Fetching aar files...
Copied 113 AARs

```

**Comment by pswaminathan** [comment]
@Morfly thoughts here? Not being able to build targets using the plugin removes quite a bit of functionality.

To be clear: the issue I get when I uncheck "Use mobile-install" is Bazel failing with:

```
ERROR: Skipping '': the empty string is not a valid target
ERROR: the empty string is not a valid target
```

It looks like it comes from `--output_groups=+android_deploy_info ""`?

**Comment by gdier** [comment]
> @Morfly thoughts here? Not being able to build targets using the plugin removes quite a bit of functionality.
> 
> To be clear: the issue I get when I uncheck "Use mobile-install" is Bazel failing with:
> 
> ```
> ERROR: Skipping '': the empty string is not a valid target
> ERROR: the empty string is not a valid target
> ```
> 
> It looks like it comes from `--output_groups=+android_deploy_info ""`?

I have the same problem with you. How can I remove **""** here? It seems auto-appended by plugin

**Comment by gdier** [comment]
Full build command line:
```
./bazel build --tool_tag=ijwb:AndroidStudio --output_groups=+android_deploy_info "" --copt=-g --fission=no --strip=never -c dbg --curses=no --color=yes --progress_in_terminal_title=no --compilation_mode=dbg --copt=-DDEBUG --copt=-fdebug-compilation-dir --copt=/proc/self/cwd --build_event_binary_file=/var/folders/ks/trkpb5v50l3cz22x2rljvpv40000gn/T/intellij-bep-37af0234-1605-408a-bfc8-dc5a9df715d4 --nobuild_event_binary_file_path_conversion //sample/android/demo:demo
```

Seems like NativeSymbolFinder::getAdditionalBuildFlags returns `"" --copt=-g ...  --copt=-DDEBUG` [Code Here](https://github.com/bazelbuild/intellij/blob/af415dc4348f9f43051f2b4da2751976be2705e4/aswb/src/com/google/idea/blaze/android/run/runner/FullApkBuildStep.java#L170)

**Comment by idanakav** [comment]
The actual issue here is that we are appending `tcp:port` to the serial number, I don't believe adb support that in the install command, at least I could not see any reference for that in the source, it may be relevant for other commands such as connect.

**Comment by idanakav** [comment]
Took a deeper look, the `tcp:port` may be relevant with an active tunnel config, otherwise, we should use a different way to specify the adb server port.

**Comment by mai93** [comment]
This should be fixed with https://github.com/bazelbuild/intellij/commit/32e2646a805fd0b82a11e6f7ee283f11118177e5 available starting from ASwB plugin version [2023.04.18](https://plugins.jetbrains.com/plugin/9185-bazel-for-android-studio/versions/beta/325815)
